src/GeoJson/Feature/Feature.php

```diff
@@ -36,17 +38,15 @@ class Feature extends GeoJson
 
     /**
      * @param int|string|null $id
-     * @param CoordinateResolutionSystem|BoundingBox $arg,...
+     * @param CoordinateReferenceSystem|BoundingBox $args
      */
-    public function __construct(?Geometry $geometry = null, ?array $properties = null, $id = null)
+    public function __construct(?Geometry $geometry = null, ?array $properties = null, $id = null, ... $args)
```

Comment:
`func_get_args()` was only used in the original code to avoid making CoordinateReferenceSystem and BoundingBox both positional parameters.

I realize this is outside the scope of this PR, but per your comment in https://github.com/jmikola/geojson/pull/33#issuecomment-1018947287 about potentially removing CoordinateReferenceSystem in 2.0, do you think it'd ultimately make sense to replace this with a single, optional BoundingBox parameter or bump the PHP 8.0 and solicit BoundingBox as a [named argument](https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments)?

Reply:
That would really be nice (although named parameters are happening in userland anyway and I wouldn't use them in the library, but this might be a conversation in another thread 🙈)!

While I read this and the other comments you made, I would like to extend my comment above it and say that I would suggest a 1.x release without breaking changes (we addressed some of them), but with deprecation notices, e.g. on the CoordinateReferenceSystem classes.

We could then aim for a 2.x release with the breaking changes (in fewer commits to make them easier to follow). That release could come shortly after.

This would give users the chance to benefit from the updated code without having to upgrade their dependencies immediately. Do you think that makes sense?

Reply:
> I would suggest a 1.x release without breaking changes (we addressed some of them), but with deprecation notices, e.g. on the CoordinateReferenceSystem classes.

I concur. I'd want to merge this to complete a 1.1.0 release (alluded to in https://github.com/jmikola/geojson/issues/34) and then invite you to submit any other PRs targeting a 2.0 release.

Adding deprecation notices for CoordinateReferenceSystem seems reasonable as well. Do you think that would best be done as triggering `E_USER_DEPRECATED` from the constructors, or would you also consider doing so during JSON unserialization?

Reply:
I've seen projects where any kind of `E_USER_*` messages that weren't there before broke applications either during CI builds or on live environments if the prod server had inappropriate `error_reporting` settings. Because of that my first impulse was to say "I wouldn't" so that the wrath of angry users doesn't hit you here 😅 

On the other hand, if the classes were only to get a `@deprecated since 1.1` annotation and a corresponding release not, most people wouldn't be aware of the deprecation unless the `E_USER_DEPRECATED` is triggered during unserialization. 🤔 

My feeling is still to only add the deprecation annotations - the worst that can happen is that the `crs` fields disappear once someone upgrades to a 2.0 release, and if they then complain about it, it would be easy to say "Haven't you read the release notes?" - I don't know if that's too mean, though 😬

Reply:
Noted. I'm not too worried about user complaints here, but if `E_USER_DEPRECATED` were to be introduced I think it'd make sense to only do it in the constructor and not during JSON deserialization to be less disruptive.

That said, I'm totally OK with deprecating the class in 1.1 (via documentation) and then simply removing it in 2.0. This could be announced in release notes for both.